src/host.ps1

```diff
@@ -1,43 +1,73 @@
+$script:XO_HOST_FIELDS = "id,uuid,name_label,name_description,power_state,memory,address,hostname,version,productBrand,build,cpus,startTime,tags,bios_strings,software_version,cpu_info,CPUs"
 
 function ConvertTo-XoHostObject {
     <#
     .SYNOPSIS
         Convert a host object from the API to a PowerShell object.
     .DESCRIPTION
         Convert a host object from the API to a PowerShell object with proper properties and types.
-    .PARAMETER InputObject
+    .PARAMETER HostData
         The host object from the API.
     #>
     [CmdletBinding()]
     [OutputType("XoPowershell.Host")]
     param (
-        [Parameter(Mandatory, ValueFromPipeline)]
-        [object]$InputObject
+        [Parameter(Mandatory, Position = 0)]
+        $HostData
     )
     
-    process {
-        $resultObject = [PSCustomObject]@{
-            PSTypeName = "XoPowershell.Host"
-            HostUuid = $InputObject.id
-            Description = $InputObject.name_description
-            Label = $InputObject.name_label
-            MemoryFree = [int64]$InputObject.memory.free
-            MemoryTotal = [int64]$InputObject.memory.total
-            Cpus = $InputObject.cpus.cores
-            Power = $InputObject.power_state
-            StartTime = $InputObject.startTime
-            Tags = $InputObject.tags
-            Address = $InputObject.address
-            Bios = $InputObject.bios_strings.bios_vendor
-            CPUModel = $InputObject.cpu_info.modelname
-            Build = $InputObject.software_version.build_number
-            Version = $InputObject.software_version.product_version
-            ProductBrand = $InputObject.software_version.product_brand
+    if ($HostData -is [string]) {
+        try {
+            Write-Verbose "HostData is a string, attempting to convert from JSON"
+            $HostData = $HostData | ConvertFrom-Json -AsHashtable
+        } catch {
+            Write-Error "Failed to convert host data from JSON string: $_"
+            return $null
+        }
+    }
+    
+    Write-Verbose "Processing host data: name_label=$($HostData.name_label), id=$($HostData.id), uuid=$($HostData.uuid), address=$($HostData.address)"
+
+    $hostObj = [PSCustomObject]@{
+        PSTypeName  = "XoHost"
+        HostUuid    = $HostData.id ?? $HostData.uuid ?? ""
+        Name        = $HostData.name_label ?? "Unknown"
+        Address     = $HostData.address ?? "Unknown"
+        PowerState  = $HostData.power_state ?? "Unknown"
+        MemoryFree  = 0
+        MemoryTotal = 0
+        Cpus        = 0
+        Description = $HostData.name_description
+        StartTime   = $HostData.startTime
+        Tags        = $HostData.tags
+        Version     = $HostData.version
+        ProductBrand = $HostData.productBrand
+    }
+
+    if ($HostData.ContainsKey('cpus') -and $HostData['cpus'] -and $HostData['cpus'].ContainsKey('cores')) {
+        $hostObj.Cpus = $HostData['cpus']['cores']
```

Comment:
When does `CPUs.cores` exist? My JSON output only has `CPUs.max` and `CPUs.number`.